### ISSUE TITLE
[RFC] Passing hyperparameters to solvers (but also for i.e. dataset)

### ISSUE THREAD

**Issue by jeandut** [benchopt/benchopt#574]
When running with the CLI a solver with a specific set of hyperparameters (say a learning rate or the l1_ratio.) Currently BenchOpt syntax for the CLI is a bit heavy handed:
```
benchopt run -s MySolver[learning_rate=0.1, l1_ratio=3.]
```
Brackets [, ] syntax is not super convenient in addition in zsh for instance it might throw:
```
zsh: no matches found: MySolver[learning_rate=0.1, l1_ratio=3.]
```
Because it uses brackets (solution is to add quotes).
Config files can help to a certain extent but it would be cool if the CLI allowed argparse-like syntax for:
```
benchopt run -s MySolver --learning_rate 0.1 --l1_ratio=3.
```
Because everyone is kinda used to the above (see e.g. [click](https://click.palletsprojects.com/en/8.1.x/) or [argparse](https://docs.python.org/3/library/argparse.html) or [fire](https://github.com/google/python-fire))
Solver hyperparameters CLI is dynamically generated from the parameters attribute of the given solver.
Ofc if one gives several solvers it complicates stuff.
I am nitpicking a bit maybe current system is much safer and both syntax are close. What do you think ?

**Comment by jeandut** [comment]
Also the syntax for the yaml config follows the CLI, aka MySolver[learning_rate=0.1, l1_ratio=3.], which does not follow [mlcollections](https://github.com/google/ml_collections) standard of using nested dicts (also it would allow nested params aka passing a function with its args to a solver)
